tests/test_pubsub.py

```diff
@@ -545,3 +545,27 @@ def test_get_message_with_timeout_returns_none(self, r):
         p.subscribe('foo')
         assert wait_for_message(p) == make_message('subscribe', 'foo', 1)
         assert p.get_message(timeout=0.01) is None
+
+
+class TestPubSubWorkerThread(object):
+    def setup_method(self, method):
+        self.message = None
+
+    def message_handler(self, message):
+        self.message = message
+
+    def test_run_in_thread(self, r):
```

Comment:
Interesting thing: this test passes, but we leave around some error text in the test output:

```
  /redis-py/tests/test_commands.py:378: ResourceWarning: unclosed <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('172.18.0.7', 35596)>

    r.ping()
```

The test seems to be using this object as it's designed to be used, so I'm not sure this is important.